### PR TITLE
Added and implemented logits processor function list and stop criteria function list.

### PR DESCRIPTION
**Pull request by Maximilian-Winter**
I added and implemented logits processor function list and stop criteria function list for use with microsoft guidance.

### CODE REVIEW

**Comment by mmagnesium** [comment]
This is going to be really helpful. Do you need any help with this?

**Comment by Maximilian-Winter** [comment]
@mmagnesium I have implemented them basically
